test/transcoding/BuildNDRange_2.ll

```diff
@@ -76,42 +76,41 @@ target triple = "spir64-unknown-unknown"
 @test_ndrange_2D3D.lsize2 = private constant [2 x i64] [i64 1, i64 1], align 8
 @test_ndrange_2D3D.lsize3 = private constant [3 x i64] [i64 1, i64 1, i64 1], align 8
 
-
 ; Function Attrs: nounwind
 define spir_func void @test_ndrange_2D3D() #0 {
 entry:
   %lsize2 = alloca [2 x i64], align 8
   %tmp = alloca %struct.ndrange_t, align 8
   %lsize3 = alloca [3 x i64], align 8
   %tmp3 = alloca %struct.ndrange_t, align 8
-  %0 = bitcast [2 x i64]* %lsize2 to i8*
-  call void @llvm.memcpy.p0i8.p0i8.i64(i8* align 8 %0, i8* align 8 bitcast ([2 x i64]* @test_ndrange_2D3D.lsize2 to i8*), i64 16, i1 false)
-  %arraydecay = getelementptr inbounds [2 x i64], [2 x i64]* %lsize2, i64 0, i64 0
-  call spir_func void @_Z10ndrange_2DPKm(%struct.ndrange_t* sret(%struct.ndrange_t) %tmp, i64* %arraydecay) #2
-  %1 = bitcast [3 x i64]* %lsize3 to i8*
-  call void @llvm.memcpy.p0i8.p0i8.i64(i8* align 8 %1, i8* align 8 bitcast ([3 x i64]* @test_ndrange_2D3D.lsize3 to i8*), i64 24, i1 false)
-  %arraydecay2 = getelementptr inbounds [3 x i64], [3 x i64]* %lsize3, i64 0, i64 0
-  call spir_func void @_Z10ndrange_3DPKm(%struct.ndrange_t* sret(%struct.ndrange_t) %tmp3, i64* %arraydecay2) #2
+  %0 = bitcast ptr %lsize2 to ptr
+  call void @llvm.memcpy.p0.p0.i64(ptr align 8 %0, ptr align 8 @test_ndrange_2D3D.lsize2, i64 16, i1 false)
+  %arraydecay = getelementptr inbounds [2 x i64], ptr %lsize2, i64 0, i64 0
+  call spir_func void @_Z10ndrange_2DPKm(ptr sret(%struct.ndrange_t) %tmp, ptr %arraydecay) #3
+  %1 = bitcast ptr %lsize3 to ptr
+  call void @llvm.memcpy.p0.p0.i64(ptr align 8 %1, ptr align 8 @test_ndrange_2D3D.lsize3, i64 24, i1 false)
+  %arraydecay2 = getelementptr inbounds [3 x i64], ptr %lsize3, i64 0, i64 0
+  call spir_func void @_Z10ndrange_3DPKm(ptr sret(%struct.ndrange_t) %tmp3, ptr %arraydecay2) #3
   ret void
 }
 
-; Function Attrs: nounwind
-declare void @llvm.memcpy.p0i8.p0i8.i64(i8* nocapture, i8* nocapture readonly, i64, i1) #2
-
-declare spir_func void @_Z10ndrange_2DPKm(%struct.ndrange_t* sret(%struct.ndrange_t), i64*) #1
+declare spir_func void @_Z10ndrange_2DPKm(ptr sret(%struct.ndrange_t), ptr) #1
 
-declare spir_func void @_Z10ndrange_3DPKm(%struct.ndrange_t* sret(%struct.ndrange_t), i64*) #1
+declare spir_func void @_Z10ndrange_3DPKm(ptr sret(%struct.ndrange_t), ptr) #1
 
 ; Function Attrs: nounwind
 define spir_func void @test_ndrange_const_2D3D() #0 {
 entry:
   %tmp = alloca %struct.ndrange_t, align 8
   %tmp1 = alloca %struct.ndrange_t, align 8
-  call spir_func void @_Z10ndrange_2DPKm(%struct.ndrange_t* sret(%struct.ndrange_t) %tmp, i64* getelementptr inbounds ([2 x i64], [2 x i64]* @test_ndrange_2D3D.lsize2, i64 0, i64 0)) #2
```

Comment:
Basically here with opaque pointers GEP is being eliminated and our implementation of ndrange handle relies on it. I've attempted to add artificial ptr GEP 0, 0, but it seems like LLVM ignores it leading to the same assertion to fire.
```
#5  0x0000000000552153 in llvm::cast<llvm::GEPOperator, llvm::Value> (Val=0x5a05750)
    at LLVM/llvm-project/llvm/include/llvm/Support/Casting.h:578
#6  0x0000000000522a83 in SPIRV::getScalarOrArray (V=0x5a05750, Size=2, Pos=0x59fadd0)
    at SPIRV-LLVM-Translator/lib/SPIRV/SPIRVUtil.cpp:1442
#7  0x00000000006f64e2 in operator() (__closure=0x7fffffffbd10, V=0x5a05750)
    at SPIRV-LLVM-Translator/lib/SPIRV/OCLToSPIRV.cpp:443
#8  0x000000000070119c in SPIRV::BuiltinCallMutator::mapArg<SPIRV::OCLToSPIRVBase::visitCallNDRange(llvm::CallInst*, llvm::StringRef)::<lambda(llvm::Value*)> >(unsigned int, struct {...}) (this=0x7fffffffbe30, Index=1, Func=...)
    at SPIRV-LLVM-Translator/lib/SPIRV/SPIRVBuiltinHelper.h:226
#9  0x00000000006f6757 in SPIRV::OCLToSPIRVBase::visitCallNDRange (this=0x5a06b98, CI=0x59fadd0, DemangledName=...)

```

Reply:
I'll have to take a look at this, there's another test that crashes in reverse translation as well that I found when mass-converting tests.